realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -184,25 +185,44 @@ private RealmQuery(BaseRealm realm, OsList osList, Class<E> clazz) {
             // TODO Queries on primitive lists are not yet supported
             this.schema = null;
             this.table = null;
-            this.osList = null;
+            this.osCollection = null;
             this.query = null;
         } else {
             //noinspection unchecked
             this.schema = realm.getSchema().getSchemaForClass((Class<? extends RealmModel>) clazz);
             this.table = schema.getTable();
-            this.osList = osList;
+            this.osCollection = osList;
             this.query = osList.getQuery();
         }
     }
 
+    RealmQuery(BaseRealm realm, OsSet osSet, Class<E> clazz) {
+        this.realm = realm;
+        this.clazz = clazz;
+        this.forValues = !isClassForRealmModel(clazz);
+        if (forValues) {
+            // TODO Queries on primitive lists are not yet supported
```

Comment:
I can see that it is just replicating the above pattern, but wouldn't it make more sense to throw an `UnsupportedOperationException` instead? I guess it is already guarded as it will never be constructed for non RealmModel types. Then I guess it makes even more sense to catch that the query is not constructed properly when/if trying to lift the restriction later.